src/components/Map.js

```diff
@@ -210,7 +210,7 @@ export const Map = React.memo(({ layers, sideBarCollapsed }) => {
             break;
         }
 
-        layerControl.addOverlay(newMapOverlays[ukey], ukey);
+        layerControl.addOverlay(newMapOverlays[ukey], name);
```

Comment:
i'm still showing name to user :thinking: 
changed this intentionally because having multiple layers with the same name (e.g. "New Layer") was leading to issues with layers deletion. if i remember correctly the last created layer was getting deleted instead of the one i wanted to delete

Reply:
Ok I see, I'll experiment layer deletion, I think we can remove a layer without using its name (?), otherwise I'll rollback this update.